### ISSUE TITLE
`no-whitespace-before-property`: False positive in parenthesized indexed access type

### ISSUE THREAD

**Issue by gebsh** [eslint-stylistic/eslint-stylistic#1097]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

```ts
const arr = [ 0, 1 ] as const;

type Test = ( typeof arr )[ number ];
```

In the above code, the rule `no-whitespace-before-property` incorrectly reports space between `typeof arr` and `)`. Furthermore, running ESLint with `--fix` introduces a syntax error by changing this code to the following:

```ts
const arr = [ 0, 1 ] as const;

type Test = ( typeof arr[ number ];
```

I suspect this is a regression introduced in #1046.

### Reproduction

https://stackblitz.com/edit/github-cqecethi

### System Info

```shell
System:
  OS: macOS 26.2
  CPU: (14) arm64 Apple M4 Pro
  Memory: 2.51 GB / 24.00 GB
  Shell: 0.109.1 - /opt/homebrew/bin/nu
Binaries:
  Node: 24.12.0 - /opt/n/bin/node
  npm: 11.6.2 - /opt/n/bin/npm
  pnpm: 10.18.2 - /opt/homebrew/bin/pnpm
npmPackages:
  eslint: ^9.39.2 => 9.39.2
```

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
